src/map/packets/quest_mission_log.h

```diff
@@ -25,66 +25,73 @@
-#define COA_COMPLETE		0x0108
-
-//Mission Log Types
-#define MISS_COMPLETE		0xD0
-#define MISS_CURRENT		0xFFFF
-
-// MISS_CURRENT	0xFF исключая Add-on Scenarios
-
-#define EXP_MISS_COMPLETE               0xD8
-#define EXP_MISS_CURRENT                0x80
-
-#define CAMPAIGN_MISSION_UN             0x30
-#define CAMPAIGN_MISSION_DEUX           0x38
-
-// Mission Logs
-#define MISSION_SANDORIA                0x0B
-#define MISSION_BASTOK                  0x0C
-#define MISSION_WINDURST                0x0D
-#define MISSION_ZILART                  0x0E
-#define MISSION_TOAU                    0x0F
-#define MISSION_WOTG                    0x10
-#define MISSION_COP                     0x11
-
-#define MISSION_ASSAULT                 0x12
-
-#define MISSION_CAMPAIGN                0x13
-#define MISSION_CAMPAIGN2               0x14
-
-#define MISSION_ACP                     0x15
-#define MISSION_AMK		                0x16
-#define MISSION_ASA		                0x17
-
-#define MISSION_SOA					    0x18
-#define MISSION_ROV                     0x19
+// Quest/Mission Area Identifiers
+#define QUESTS_SANDORIA      0
+#define QUESTS_BASTOK        1
+#define QUESTS_WINDURST      2
+#define QUESTS_JEUNO         3
+#define QUESTS_OTHER         4
+#define QUESTS_OUTLANDS      5
+#define QUESTS_TOAU          6
+#define QUESTS_WOTG          7
+#define QUESTS_ABYSSEA       8
+#define QUESTS_ADOULIN       9
+#define QUESTS_COALITION    10
+
+#define MISSION_SANDORIA     0
+#define MISSION_BASTOK       1
+#define MISSION_WINDURST     2
+#define MISSION_ZILART       3
+#define MISSION_TOAU         4
+#define MISSION_WOTG         5
+#define MISSION_COP          6
+#define MISSION_ASSAULT      7
+#define MISSION_CAMPAIGN     8
+#define MISSION_ACP          9
+#define MISSION_AMK         10
+#define MISSION_ASA         11
+#define MISSION_SOA         12
+#define MISSION_ROV         13
+
+// Log Types
+enum LOG_TYPE
+{
+    LOG_QUEST_CURR = 1,
+    LOG_QUEST_COMP = 2,
+    LOG_MISS_CURR = 3,
+    LOG_MISS_COMP = 4,
+    LOG_CAMPAIGN2 = 5,
+};
+
+// Quest Log Packet Values
+static const std::unordered_map<uint8, std::pair<uint16, uint16>> questPacketBytes =
+{
+    // Quest Log ID   , Current, Complete
+    { QUESTS_SANDORIA , {0x50  , 0x90   }},
+    { QUESTS_BASTOK   , {0x58  , 0x98   }},
+    { QUESTS_WINDURST , {0x60  , 0xA0   }},
+    { QUESTS_JEUNO    , {0x68  , 0xA8   }},
+    { QUESTS_OTHER    , {0x70  , 0xB0   }},
+    { QUESTS_OUTLANDS , {0x78  , 0xB8   }},
+    { QUESTS_TOAU     , {0x80  , 0xC0   }},
+    { QUESTS_WOTG     , {0x88  , 0xC8   }},
+    { QUESTS_ABYSSEA  , {0xE0  , 0xE8   }},
+    { QUESTS_ADOULIN  , {0xF0  , 0xF8   }},
+    { QUESTS_COALITION, {0x0100, 0x0108 }},
+};
+
+// Mission Log Packet Bytes
+#define MISS_COMPLETE       0xD0
+#define MISS_CURRENT        0xFFFF
+#define EXP_MISS_COMPLETE   0xD8
+#define EXP_MISS_CURRENT    0x80
+#define ASSAULT_COMPLETE    0xC0
+#define CMPGN_MISS_UN       0x30
+#define CMPGN_MISS_DEUX     0x38
```

Comment:
you could replace UN and DEUX with one and two while you're at it..

On Tue, Aug 9, 2016 at 6:25 AM, Tahir Akhlaq <notifications@github.com>
wrote:

> In src/map/packets/quest_mission_log.h
> <https://github.com/DarkstarProject/darkstar/pull/3292#discussion_r74047461>
> :
>
> > +    { QUESTS_OUTLANDS , {0x78  , 0xB8   }},
> > +    { QUESTS_TOAU     , {0x80  , 0xC0   }},
> > +    { QUESTS_WOTG     , {0x88  , 0xC8   }},
> > +    { QUESTS_ABYSSEA  , {0xE0  , 0xE8   }},
> > +    { QUESTS_ADOULIN  , {0xF0  , 0xF8   }},
> > +    { QUESTS_COALITION, {0x0100, 0x0108 }},
> > +};
> > +
> > +// Mission Log Packet Bytes
> > +#define MISS_COMPLETE       0xD0
> > +#define MISS_CURRENT        0xFFFF
> > +#define EXP_MISS_COMPLETE   0xD8
> > +#define EXP_MISS_CURRENT    0x80
> > +#define ASSAULT_COMPLETE    0xC0
> > +#define CMPGN_MISS_UN       0x30
> > +#define CMPGN_MISS_DEUX     0x38
>
> CAMPAIGN_MISSION_DEUX etc
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/3292/files/9da146335e7e155e626ce4ad0aeb62f44a6b328e#r74047461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGI_wzBKRmy-_4fRa3i3CC0l4Q2Qw1Aks5qeHHTgaJpZM4JR0YG>
> .
>